### PR TITLE
Add proper delete handling for threads and posts

### PR DESCRIPTION
**Pull request by michalsn**
This PR adds a proper delete handling for individual Threads and Posts - with statistics calculation.

### CODE REVIEW

**Review by lonnieezell — APPROVED** [review]

File: app/Concerns/ImpactsCategoryCounts.php [hunk 1]
```diff
@@ -29,19 +31,60 @@ protected function incrementThreadCount(array $data): bool|array
...
+        // The $data['id'] is always an array on delete
+        $threads = $this->allowCallbacks(false)->onlyDeleted()->find($data['id']);
+
+        foreach ($threads as $thread) {
+            $postModel = model(PostModel::class);
+            // Soft-delete all the posts for this thread
+            $postModel->builder()
+                ->update(
+                    ['deleted_at' => $thread->deleted_at],
+                    ['thread_id' => $thread->id, 'deleted_at' => null]
```
Comment (datamweb) [inline app/Concerns/ImpactsCategoryCounts.php:52]: Shouldn't `deleted_at`  be added to `$allowedFields`?
https://github.com/lonnieezell/forum-example/blob/b4010a8c72a33e64cb9a93c9f656cb26fa387beb/app/Models/PostModel.php#L25-L27

Comment (michalsn) [reply app/Concerns/ImpactsCategoryCounts.php:52]: For now, I don't think so. There is no scenario where we will be updating this field with a model.
All stats-related stuff is made with the use of a builder instance.

**Review by datamweb — APPROVED** [review]

File: app/Concerns/ImpactsCategoryCounts.php (hunk 1 above)
Comment (datamweb) [reply app/Concerns/ImpactsCategoryCounts.php:52]: Okay,

**Comment by datamweb** [comment]
@michalsn thank you!

**Pull request merged**